include/jemalloc/internal/hpa_opts.h

```diff
@@ -7,8 +7,53 @@
 /*
  * This file is morally part of hpa.h, but is split out for header-ordering
  * reasons.
+ *
+ * All of these hpa_shard_opts below are experimental. We are exploring more
+ * efficient packing, hugifying, and purging approaches to make efficient
+ * trade-offs between CPU, memory, latency, and usability. This means all of
+ * them are at the risk of being deprecated and corresponding configurations
+ * should be updated once the final version settles.
  */
 
+/*
+ * This enum controls when jemalloc hugifies/dehugifies pages.  Each mode may be
+ * more suitable depending on deployment environments.
+ *
+ * hpa_hugify_style_none
+ * This style only makes sense when deploying on systems where THP are enabled
+ * in 'always' mode.  With this style, you most likely want to have no purging
```

Comment:
The cases below explain what the option does.  This should, too.  Something along the lines of, "in this mode, JEMalloc delegates the responsibility of promoting and demoting page sizes to the operating system and will avoid doing things that can mess that up."